lib/src/wiggle_abi/req_impl.rs

```diff
@@ -556,13 +556,13 @@ impl FastlyHttpReq for Session {
 
         // asynchronously send the request
         let pending_req =
-            PendingRequest::spawn(upstream::send_request(req, backend, self.tls_config()));
+            PeekableTask::spawn(upstream::send_request(req, backend, self.tls_config()));
 
         // return a handle to the pending request
-        Ok(self.insert_pending_request(pending_req))
+        Ok(self.insert_pending_request(pending_req.await))
```

Comment:
I think it'd look more natural to put the `.await` directly after the `spawn` on line 559